File: src/main/java/com/example/chatService/repository/GroupMessageRepository.java
package com.example.chatService.repository;

import com.example.chatService.entity.GroupMessageEntity;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface GroupMessageRepository extends JpaRepository<GroupMessageEntity, Long> {

    @Query("""
        SELECT m FROM GroupMessageEntity m
        WHERE m.roomId = :roomId
        ORDER BY m.createdAt DESC
    """)
    List<GroupMessageEntity> findRecent(
            @Param("roomId") String roomId,
            Pageable pageable
    );
}

File: src/main/java/com/example/chatService/repository/GroupMessageRepository.kt
package com.example.chatService.repository

import com.example.chatService.entity.GroupMessageEntity
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface GroupMessageRepository : JpaRepository<GroupMessageEntity, Long> {
    @Query(
        """
        SELECT m FROM GroupMessageEntity m
        WHERE m.roomId = :roomId
        ORDER BY m.createdAt DESC
    """,
    )
    fun findRecent(@Param("roomId") roomId: String, pageable: Pageable): List<GroupMessageEntity>
}
